docs/java-rest/high-level/watcher/execute-watch.asciidoc

```diff
@@ -0,0 +1,88 @@
+--
+:api: execute-watch
+:request: ExecuteWatchRequest
+:response: ExecuteWatchResponse
+--
+[id="{upid}-{api}"]
+=== Execute Watch API
+
+The execute watch API allows clients to immediately execute a watch, either
+one that has been previously added via the
+{ref}/put-watch.html[Put Watch API] or inline as part of the request.
+
+[id="{upid}-{api}-request-by-id"]
+==== Execute by id
+
+Submit the following request to execute a previously added watch:
+
+["source","java",subs="attributes,callouts,macros"]
+---------------------------------------------------
+include-tagged::{doc-tests-file}[x-pack-execute-watch-by-id]
+---------------------------------------------------
+<1> Alternative input for the watch to use in json format
+<2> Set the mode for action "action1" to SIMULATE
+<3> Record this execution in watcher history
+<4> Execute the watch regardless of the watch's condition
+<5> Set the trigger data for the watch in json format
+<6> Enable debug mode
+
+[id="{upid}-{api}-response-by-id"]
+==== Execute by id Response
+
+The returned `Response` contains details of the execution:
+
+["source","java",subs="attributes,callouts,macros"]
+---------------------------------------------------
+include-tagged::{doc-tests-file}[x-pack-execute-watch-by-id-response]
+---------------------------------------------------
+<1> The record ID for this execution
+<2> The execution response as a java `Map`
+<3> Extract information from the response map using `ObjectPath`
+
+[id="{upid}-{api}-response-by-id-async"]
+==== Asynchronous execution by id
```

Comment:
the async stuff can be pulled out assuming its different from the boilerplate stuff. See `include::../execution.asciidoc[]` that is included in other docs pages.

Reply:
I don't think that will work here because there are two types of request and response, by ID and inline.